client/control/actions/FilterActions.js

```diff
@@ -0,0 +1,113 @@
+import {
+  makeApiRequest,
+  recipesNeedFetch,
+  recipesReceived,
+  filtersReceived,
+} from 'control/actions/ControlActions';
+
+import {
+  getFilterParamString,
+} from 'control/selectors/FiltersSelector';
+
+const LOAD_FILTERS = 'LOAD_FILTERS';
+const SET_FILTER = 'SET_FILTER';
+const SET_TEXT_FILTER = 'SET_TEXT_FILTER';
+const SET_ALL_FILTERS = 'SET_ALL_FILTERS';
+
+
+/**
+ * Load list of possible filters from remote API.
+ * This is stored in the `filters` reducer, and
+ * later used to populate relevant RecipeFilters components.
+ */
+function loadFilters() {
+  return dispatch =>
+    dispatch(makeApiRequest('fetchFilters'))
+      .then(recipes => dispatch(filtersReceived(recipes)));
+}
+
+/**
+ * Given an option and its parent group, update the
+ * filter state based on the `isEnabled` prop
+ *
+ * @param  {Object}  group     Group the filter belongs to
+ * @param  {Object}  option    Option that was affected
+ * @param  {Boolean} isEnabled Is the option selected?
+ */
+function selectFilter({ group, option, isEnabled }) {
+  return dispatch => {
+    dispatch({
+      type: group.value === 'text' ? SET_TEXT_FILTER : SET_FILTER,
+      group,
+      option,
+      isEnabled,
+    });
+  };
+}
+
+
+/**
+ * Detects activated filters, creates the URL param string,
+ * and queries API for a filtered list based on params.
+ */
+function loadFilteredRecipes() {
+  return (dispatch, getState) => {
+    dispatch(recipesNeedFetch());
+
+    const filterParams = getFilterParamString(getState().filters);
+
+    return dispatch(makeApiRequest('fetchFilteredRecipes', filterParams))
+      .then(recipes => dispatch(recipesReceived(recipes, filterParams)));
+  };
+}
+
+/**
+ * Dispatches a SET_ALL_FILTERS event.
+ * If called without a state param,
+ * the filters reducer will reset the state
+ * to its initialState.
+ *
+ * The difference between SET_ALL_FILTERS and LOAD_FILTERS
+ * is that LOAD_ parses data from the backend before
+ * inserting into the store. SET_ALL assumes that
+ * the filters are already parsed, and the _app_ is
+ * manipulating the existing store.
```

Comment:
I think this is going to be confusing in the long run. I think a better approach would be to only have `SET_ALL_FILTERS`, and make whatever dispatches `LOAD_FILTERS` do the parsing and expansion there, instead of doing it in the reducer. If I'm understanding the flow correctly, this would make dealing with server formatting and actually making the request close together, instead of being spread across the reducers and action creators.